lmfit/parameter.py

```diff
@@ -688,13 +688,14 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         """Set state for pickle."""
-        (self.name, self.value, self.vary, self.expr, self.min, self.max,
+        (self.name, _value, self.vary, self.expr, self.min, self.max,
          self.brute_step, self.stderr, self.correl, self.init_value,
          self.user_data) = state
         self._expr_ast = None
         self._expr_eval = None
         self._expr_deps = []
         self._delay_asteval = False
+        self.value = _value
```

Comment:
Again, I am missing something, what does this change net us?

Reply:
@Tillsten  Ah, but here it makes a difference because `value` is a property that now causes the bounds to be forced.   So when setting `self.value = _val` we would be well-off to have the bounds and any expressions correctly defined.

Reply:
I would think that the old assignment from the tuple should also trigger the property setter?

Reply:
@Tillsten yes, it was.  But it was triggering the property before all the other attributes were set.   So, in this PR it sets all the attributes (bounds, expression, expression evaluator) and then triggers the property.